src/java/org/apache/cassandra/service/BatchlogResponseHandler.java

```diff
@@ -31,6 +31,8 @@ public class BatchlogResponseHandler<T> extends AbstractWriteResponseHandler<T>
     AbstractWriteResponseHandler<T> wrapped;
     BatchlogCleanup cleanup;
     protected volatile int requiredBeforeFinish;
+    // Defer cleanup until `maybeAckMutationForCleanup` is called
+    private boolean deferCleanup = false;
```

Comment:
I ended up refactoring this to remove `deferCleanup` entirely since `Cleanup` can be interacted with more directly to account for the actual number of wrappers that need to be created + 1 ack for the Accord txn.